plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/BulkIndexByScrollResponse.java

```diff
@@ -28,157 +28,88 @@
 import org.elasticsearch.common.xcontent.XContentBuilder;
 
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
 
-import static java.lang.Math.min;
-import static java.util.Collections.unmodifiableList;
-import static org.elasticsearch.action.search.ShardSearchFailure.readShardSearchFailure;
+import static java.util.Objects.requireNonNull;
 
 /**
  * Response used for actions that index many documents using a scroll request.
  */
 public class BulkIndexByScrollResponse extends ActionResponse implements ToXContent {
-    static final String TOOK_FIELD = "took";
-    static final String UPDATED_FIELD = "updated";
-    static final String BATCHES_FIELD = "batches";
-    static final String VERSION_CONFLICTS_FIELD = "version_conflicts";
-    static final String NOOPS_FIELD = "noops";
-    static final String FAILURES_FIELD = "failures";
-
     private long took;
-    private long updated;
-    private int batches;
-    private long versionConflicts;
-    private long noops;
-    private List<Failure> indexingFailures;
-    private List<ShardSearchFailure> searchFailures;
+    private BulkByScrollTask.Status status;
 
     public BulkIndexByScrollResponse() {
     }
 
-    public BulkIndexByScrollResponse(long took, long updated, int batches, long versionConflicts, long noops,
-            List<Failure> indexingFailures, List<ShardSearchFailure> searchFailures) {
+    public BulkIndexByScrollResponse(long took, BulkByScrollTask.Status status) {
         this.took = took;
-        this.updated = updated;
-        this.batches = batches;
-        this.versionConflicts = versionConflicts;
-        this.noops = noops;
-        this.indexingFailures = indexingFailures;
-        this.searchFailures = searchFailures;
+        this.status = requireNonNull(status);
     }
 
     public long getTook() {
         return took;
     }
 
+    protected BulkByScrollTask.Status getStatus() {
+        return status;
+    }
+
+    public String getWriteableName() {
```

Comment:
I was going to make the status contain the response but that was more painful than going the other way. This was just left over.